### DISCUSSION TITLE
Why is the min refresh interval this high?

### DISCUSSION THREAD

**Discussion by Grizzler89** [kodebach/hacs-idm-heatpump#118] [Q&A]
Hello.

The minimum refresh interval is currently 1 minute. That seems quite high to me.

The web interface updates the values every 5 seconds. 

Is there a technical reason why the refresh via modbus has to be so infrequent?

Best regards

**Comment by kodebach** [comment]
A user reported the got the info from IDM that the refresh interval should not be lower than 60 seconds (see https://github.com/kodebach/hacs-idm-heatpump/issues/16).

Also depending on your Home Assistant machine and network conditions, low refresh interval may not be feasible anyway. If you [enable debug logging](https://www.home-assistant.io/docs/configuration/troubleshooting/#enabling-debug-logging) and wait for a refresh, you should see in the logs that a refresh can be quite slow. For me updating just now took about 6s, but I don't use any room modules.

Beyond these technical reasons, I don't really see a use case for such a low refresh rate. Most sensor values won't change that frequently, if the heat pump itself even reads values from the sensor hardware this often. What would be your use case for a 5 second refresh interval in Home Assistant?

**Comment by AndyNew2** [comment]
Confirmed, have seen from time to time issues. 
I have it run on 26 seconds now stable, but it is really on the limit. Cannot below 22 seconds. 
I am about to relax more to have a stable GUI and controller.
Not sure what causes this, but by removing network connection to HA the iDM Controller and GUI recovered.

**Comment by AndyNew2** [comment]
@Grizzler89 You may want to try my new integration. This will solve your issue for a lot of entities now ... ;-) 
Hope it helps you.... 
https://github.com/AndyNew2/hacs-idm-hpweb
